trieste/utils/objectives.py

```diff
@@ -164,6 +164,85 @@ def hartmann_3(x: TensorType) -> TensorType:
 """
 
 
+def rosenbrock_4(x: TensorType) -> TensorType:
+    """
+    The Rosenbrock function, rescaled to have zero mean and unit variance over :math:`[0, 1]^4. See
+    :cite:`Picheny2013` for details.
+
+    This function (also known as the Banana function) is unimodal, however the minima
+    lies in a narrow valley.
+
+    :param x: The points at which to evaluate the function, with shape [..., 4].
+    :return: The function values at ``x``, with shape [..., 1].
+    :raise ValueError (or InvalidArgumentError): If ``x`` has an invalid shape.
+    """
+    tf.debugging.assert_shapes([(x, (..., 4))])
+
+    x = 15.0 * x - 5
+    unscaled_function = tf.reduce_sum(
+        (100.0 * (x[..., 1:] - x[..., :-1]) ** 2 + (1 - x[..., :-1]) ** 2), axis=-1, keepdims=True
+    )
+    return (unscaled_function - 3.827 * 1e5) / (3.755 * 1e5)
+
+
+ROSENBROCK_4_MINIMIZER = tf.constant([[0.4, 0.4, 0.4, 0.4]], tf.float64)
+"""
+The global minimizer for the :func:`rosenbrock_4` function, with shape [1, 4] and
+dtype float64.
+"""
+
+
+ROSENBROCK_4_MINIMUM = tf.constant([-1.01917], tf.float64)
```

Comment:
Why not have this actually evaluate the function at the minimizer?

Reply:
I think the idea is to have this independent from our code. This means that the tests can only pass if our implementation is working.